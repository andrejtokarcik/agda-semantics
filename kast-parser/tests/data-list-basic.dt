data List (a : Set 0) : Set 0 where
  Nil : List a,
  Cons : Pi x a (Pi y (List a) (List a))

data Nat : Set 0 where
  Zero : Nat,
  Succ : Pi x Nat Nat


expr = ((Lambda x (List Nat) (Cons Nat Zero x)) (Cons Nat Zero (Cons Nat (Succ
Zero) (Nil Nat))))
