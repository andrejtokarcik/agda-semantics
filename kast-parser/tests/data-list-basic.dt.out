<dbg>
    ListItem((Pi x #symVariable(14) (Pi y List #symVariable(14) List 
      #symVariable(14))))
    ListItem((Pi x #symVariable(23) (Pi y List #symVariable(23) List 
      #symVariable(23))))
    ListItem((Pi x #symVariable(29) (Pi y List #symVariable(29) List 
      #symVariable(29))))
    ListItem( List #symVariable(38) )
</dbg>
<T>
    <k>
        .K
    </k>
    <mgu>
        subst(.KList)
    </mgu>
    <data>
        Cons |-> (Pi a Set 0 (Pi x a (Pi y List a List a)))
        List |-> (Pi a Set 0 Set 0)
        Nat |-> Set 0
        Nil |-> (Pi a Set 0 List a)
        Succ |-> (Pi x Nat Nat)
        Zero |-> Nat
    </data>
    <meta>
        .Map
    </meta>
    <fun>
        expr |-> (((Lambda x List Nat Cons Nat Zero x) @ .Map)  Cons Nat 
          Zero  Cons Nat (Succ Zero) (Nil Nat)  )
    </fun>
    <tenv>
        expr |->  List Nat 
    </tenv>
</T>
